eZ/Publish/Core/Search/Legacy/Content/Handler.php

```diff
@@ -339,4 +343,69 @@ public function deleteLocation($locationId, $contentId)
     {
         // Not implemented in Legacy Storage Engine
     }
+
+    public function purgeIndex()
+    {
+        // Not implemented
+    }
+
+    /**
+     * Create search engine index.
+     *
+     * @param $bulkCount
+     * @param \eZ\Publish\SPI\Search\IndexerDataProvider $dataProvider
+     * @param \Symfony\Component\Console\Helper\ProgressBar $progress
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function createSearchIndex(
+        $bulkCount,
+        IndexerDataProvider $dataProvider,
+        ProgressBar $progress,
+        LoggerInterface $logger
+    ) {
```

Comment:
`createSearchIndex` handles an entire process of creating search index so I needed to pass here `ProgressBar` and `LoggerInterface`. I could use instead `callable` but it still would not be very clean solution. Not sure how to do it better.